src/patternfly/components/Hint/hint.scss

```diff
@@ -0,0 +1,114 @@
+.pf-c-hint {
+  --pf-c-hint__header--GridColumnWidth: auto;
+  --pf-c-hint__actions--GridColumnWidth: calc(var(--pf-global--spacer--2xl) + #{pf-size-prem(44px)});
+  --pf-c-hint--PaddingTop: var(--pf-global--spacer--lg);
+  --pf-c-hint--PaddingRight: var(--pf-global--spacer--lg);
+  --pf-c-hint--PaddingBottom: var(--pf-global--spacer--lg);
+  --pf-c-hint--PaddingLeft: var(--pf-global--spacer--lg);
+  --pf-c-hint--BackgroundColor: var(--pf-global--palette--blue-50);
+  --pf-c-hint--BorderColor: var(--pf-global--palette--blue-100);
+  --pf-c-hint--BorderWidth: var(--pf-global--BorderWidth--sm);
+  --pf-c-hint--BoxShadow: var(--pf-global--BoxShadow--sm);
+  --pf-c-hint--Color: var(--pf-global--Color--100);
+
+  // Hint Header
+  --pf-c-hint__header--FontSize: var(--pf-global--FontSize--lg);
+
+  // Hint Body
+  --pf-c-hint__body--FontSize: var(--pf-global--FontSize--md);
+  --pf-c-hint__header--body--MarginTop: var(--pf-global--spacer--md);
+
+  // Hint Footer
+  --pf-c-hint__footer--child--MarginRight: var(--pf-global--spacer--md);
+  --pf-c-hint__header--footer--MarginTop: var(--pf-global--spacer--md);
+  --pf-c-hint__body--footer--MarginTop: var(--pf-global--spacer--md);
+
+  // Hint Actions
+  --pf-c-hint__actions--MarginLeft: var(--pf-global--spacer--2xl);
+  --pf-c-hint__actions--c-dropdown--MarginTop: calc(var(--pf-global--spacer--form-element) * -1);
+
+  // Dominant modifier
+  --pf-c-hint--m-dominant--BackgroundColor: var(--pf-global--primary-color--100);
+  --pf-c-hint--m-dominant--BoxShadow: var(--pf-global--BoxShadow--md);
+  --pf-c-hint--m-dominant--Color: var(--pf-global--Color--light-100);
+
+  display: grid;
+  grid-template-columns: var(--pf-c-hint__header--GridColumnWidth) var(--pf-c-hint__actions--GridColumnWidth);
+  padding: var(--pf-c-hint--PaddingTop) var(--pf-c-hint--PaddingRight) var(--pf-c-hint--PaddingBottom) var(--pf-c-hint--PaddingLeft);
+  color: var(--pf-c-hint--Color);
+  background-color: var(--pf-c-hint--BackgroundColor);
+  border: var(--pf-c-hint--BorderWidth) solid var(--pf-c-hint--BorderColor);
+  box-shadow: var(--pf-c-hint--BoxShadow);
+
+  &.pf-m-dominant {
+    border: unset;
+
+    --pf-c-hint--Color: var(--pf-c-hint--m-dominant--Color);
+    --pf-c-hint--BackgroundColor: var(--pf-c-hint--m-dominant--BackgroundColor);
+    --pf-c-hint--BoxShadow: var(--pf-c-hint--m-dominant--BoxShadow);
+
+    .pf-c-button,
+    .pf-c-dropdown__toggle {
```

Comment:
The `pf-m-dominant` variation is a tricky one. It's effectively a theme. All of its contents need to change so that text, borders, shadows, etc, work against this background. For example, here's what it looks like with an expandable section:

<img width="280" alt="Screen Shot 2020-06-25 at 5 27 31 PM" src="https://user-images.githubusercontent.com/35148959/85801810-46e19900-b709-11ea-86b1-154716c7ca4d.png">

The button text is the same as the background. Here it is focused

<img width="281" alt="Screen Shot 2020-06-25 at 5 27 41 PM" src="https://user-images.githubusercontent.com/35148959/85801827-519c2e00-b709-11ea-9c0c-007d17bdb651.png">

To do this right, I think we need the concept of themed components. The closest we have is our existing implementation of dark theme - it doesn't work with the blue background color, but here's black for example.

<img width="897" alt="Screen Shot 2020-06-25 at 5 30 54 PM" src="https://user-images.githubusercontent.com/35148959/85802104-da1ace80-b709-11ea-8361-c559e4d3ff74.png">

Unless we want to restrict the type of content that can be placed in a hint (limit it to text, buttons, and dropdowns, for example, and style those elements with the correct colors), we may want to 

* Update the dominant variation so that it uses a normal background and everything that's placed inside of it doesn't need to be re-themed/styled so that it's visible and accessible. This is probably what I would recommend.
* Consider using the existing dark theme feature we have (would require use a different background than the dark blue), and testing to make sure our components work. This could open a can of worms since we don't use the dark theme extensively and might discover things don't work as expected.

Reply:
cc @maryshak1996 @mcarrano

Reply:
These are great points @mcoker . But before we go back and redesign this, is it possible to restrict what goes inside of here? I don't think we were intending that the hint was a general container that you can drop anything into. Thoughts @maryshak1996 ?

Reply:
@mcarrano yeah that's an option, it just currently seems like we'll run into problems when someone puts an element in the container we aren't anticipating, whether we want to support that use case or not. Like here's what it looks like if you use a link currently - 

<img width="862" alt="Screen Shot 2020-06-26 at 8 26 54 AM" src="https://user-images.githubusercontent.com/35148959/85862165-03c60b00-b787-11ea-947c-d46180fac1b3.png">

<img width="891" alt="Screen Shot 2020-06-26 at 8 27 04 AM" src="https://user-images.githubusercontent.com/35148959/85862159-0294de00-b787-11ea-898f-68d4c2e99c77.png">

Reply:
Yeah, that's a problem if we can't put link text into the message. Let me pull together a meeting to talk this through and decide how best to address.

Reply:
@mcarrano I agree, I don't think that we want too much flexibility with what goes into the hint.
@mcoker I'm a little hesitant to support a black dominant hint, since in many cases the hint will be housed close to the masthead, which is also black -- I think that it might not have the same effect when there's a black hint underneath a black masthead